### PR TITLE
Translate ch17-00

### PR DESCRIPTION
**Pull request by joeljpresent**
I translated Chapter 17-00 about OOP into French.

### CODE REVIEW

File: FRENCH/src/ch17-00-oop.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+the trade-offs of doing so versus implementing a solution using some of Rust’s
+strengths instead.
+-->
+
+La programmation orientée objet (POO) est une façon de concevoir des programmes.
+Les objets sont apparus dans Simula dans les années 1960. Ces objets ont
+influencé l'architecture de programmation d'Alan Kay dans laquelle les objets
+s'envoient des messages. Il a inventé le terme *programmation orientée objet* en
+1967 pour décrire cette architecture. Plusieurs définitions de la POO
+s'opposent ; Rust sera considéré comme orienté objet selon certaines définitions
```
Comment (Jimskapt) [inline FRENCH/src/ch17-00-oop.md]:
```suggestion
s'opposent ; Rust est considéré comme orienté objet selon certaines définitions
```

File: FRENCH/src/ch17-00-oop.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+mais pas par d'autres. Dans ce chapitre, nous examinerons certaines
+caractéristiques généralement considérées comme orientées objet et nous verrons
+comment ces caractéristiques se traduisent en code Rust idiomatique. Puis nous
```
Comment (Jimskapt) [inline FRENCH/src/ch17-00-oop.md]:
```suggestion
comment ces caractéristiques se traduisent en code Rust classique. Puis nous
```

or

```suggestion
comment ces caractéristiques se traduisent en code Rust traditionnel. Puis nous
```

File: FRENCH/src/ch17-00-oop.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+vous montrerons comment implémenter un patron de conception orienté objet en
```
Comment (Jimskapt) [inline FRENCH/src/ch17-00-oop.md]:
```suggestion
vous montrerons comment implémenter un motif de conception orienté objet en
```
(as defined in [translation-terms](https://jimskapt.github.io/rust-book-fr/translation-terms.html))

Comment (joeljpresent) [reply FRENCH/src/ch17-00-oop.md]: In French, *design patterns* are called `patrons de conception` (cf. [Wikipédia](https://fr.wikipedia.org/wiki/Patron_de_conception)).
It has nothing to do with pattern matching's *patterns*, which is what *translation-terms* refers to.

Comment (Jimskapt) [reply FRENCH/src/ch17-00-oop.md]: OK, I think you've got right 😉

**Comment by Jimskapt** [comment]
I think we will merge this Push Request when the previous chapters has been proofreaded, otherwise it will be a mess (especially in `SUMMARY.md`).

**Pull request merged**